Mathlib/Geometry/Manifold/Orientable.lean

```diff
@@ -0,0 +1,135 @@
+/-
+Copyright (c) 2024 Rida Hamadani. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rida Hamadani
+-/
+import Mathlib.Geometry.Manifold.SmoothManifoldWithCorners
+
+/-!
+# Orientable Manifolds
+
+This module defines orientable manifolds.
+
+## Main Definitions
+
+- `OrientationPreserving` : a map is orientation-preserving on a given set if the determinant of
+  its Jacobian is strictly positive on that set.
+- `orientationPreservingGroupoid` : the groupoid of partial homeos of `H` which are
+  orientation-preserving.
+- `OrientableManifold`: a type class saying that the charted space `M`, modelled on the space `H`,
+  admis an orientation.
+
+## TODO
+
+- Define orientable `0`-manifolds.
+
+-/
+
+variable {H : Type*} [NormedAddCommGroup H] [NormedSpace ℝ H]
+
+section OrientationPreserving
+
+/--
+A map is orientation-preserving on a given set if it is differentiable and the determinant of its
+Jacobian is strictly positive on that set.
+-/
+def OrientationPreserving (f : H → H) (s : Set H) : Prop :=
+  ∀ x ∈ s, 0 < (fderiv ℝ f x).det
+
```

Comment:
If I understand Lean's convention (0^0 = 1) correctly, this means 0-dimensional manifolds are orientable. (This is mathematically correct: these are simply points; an orientation of a zero-manifold simply assigns each point a value $$\pm 1$$.)
Can you add a lemma stating this, that `H` being 0-dimensional implies f is orientation-preserving?